### DISCUSSION TITLE
[question] how to combine multiple files, include only references.yml

### DISCUSSION THREAD

**Discussion by gcmartijn** [mikefarah/yq#1291] [Q&A]
H!,

Is this possible out the box with yq?
I'm trying somethings but I guess it is not ?
My goal is to create readable yml files, and than export the output to json.

I don't want to work for example with the raw id's but with an alias.
references.yml
```
reference_id:
  &tree 20000: null
  &hat 20001: null
```

something.yml
```
another_thing:
  *hat :
    - test: Nice
```

First I did try to modify the import/include/load line for line in something.yml
something.yml
```
import references.yml
include references.yml
load references.yml
import(references.yml)
include(references.yml)
load(references.yml)
another_thing:
  *hat :
    - test: Hey
```

And then do this
```
yq eval -o=json ../temp/something.yml
```

But nothing worked, then i did try (found in the manual):
```
yq eval-all -o=json '.' ../temp/*.yml
yq eval-all --tojson --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' ../temp/references.yml ../temp/something.yml 
```

Din't work either...

Do I have to make a shell script wrapper to first include the references.yml file info something.yml and than process it?

The end product don't need the content from (reference_id) references.yml , I don't need that in the json.
I only want them to fill the keys inside something.yml

**Comment by symbl-ay** [comment]
I've been doing something sort of similar to this recently. If you're alright using `make` for this, it isn't too bad. I haven't checked that these commands will actually work, particularly [the explode operator](https://mikefarah.gitbook.io/yq/operators/anchor-and-alias-operators#explode-with-merge-anchors), but they should be enough to get started.

```Makefile
json/%: references.yml yml/%
	cat $^ | yq 'explode(.)' -o j > $@.json
```

This will take each file, cat the `references.yml` file *on top of* the files selected by the user with a command like:

```bash
$: make json/*
cat references.yml yml/* | yq 'explode(.)' -o j > json/*.json
```

**Comment by gcmartijn** [comment]
I did 'solve' it by using something like this, for the moment that is working.

The problem I also had was that each file needed to end with an empty line that is the <(echo) .

```
# merge all files
# @warning each file need to end with a empty line <(echo) !
# @warning there is a order to follow
cat ../temp/i18n/language-refs.yml <(echo) ../temp/i18n/action-refs.yml > ../temp/i18n/merge.yml
cat ../temp/actions/*.yml <(echo) >> ../temp/i18n/merge.yml
cat ../temp/i18n/language-uses.yml <(echo) >> ../temp/i18n/merge.yml

# convert yaml into json and take merge the language refs and uses
yq eval-all -o=json ../temp/i18n/merge.yml | jq -r '.languagerefs * .languageuses' > ../temp/i18n/language.json
```

Inside an older code I use awk 1
```
awk 1 ../temp/actions/action-refs.yml ../temp/actions/language-refs.yml ../temp/actions/${ACTION_FILE_TEMP} > ../temp/actions/merge.yml

yq eval-all -o=json ../temp/actions/merge.yml | jq 'del(.reference, .languagerefs)' > ${ACTION_FILE_TEMP_NAME}.json
```
